### PR TITLE
Publish to npm. Closes #2

### LINKED ISSUE #2: publish to npm
my PR allows you to publish by executing one of the realease scripts.

the PR: https://github.com/micku7zu/vanilla-tilt.js/pull/1

### PR DESCRIPTION
**Pull request by bastilimbach**
@micku7zu super simple fix. I just fixed the build scripts in `package.json`. After some searching, I found out that the variable "main" in `package.json` pointed to the vanilla-tilt.js file in the `/lib` folder. This is the wrong destination, so I changed it to the usual `/dist` folder. But again, didn't work 😕. Then I saw, that there wasn't any module exported. This confused me even more because in `src/vanilla-tilt.js` you did include the ES5 `export default class VanillaTilt`. So I just ran the build again and babel did build the files correctly with `module.export` included. So I guess, someone just did forget to run the build process?! To prevent this I added a `prepublish`script to `package.json`which will automatically run on every `npm publish`.

@BrunnerLivio could you test the install with my fork before we merge? `npm install --save https://github.com/bastilimbach/vanilla-tilt.js`

### CODE REVIEW

**Comment by BrunnerLivio** [comment]
Works like it supposed to be!
Can finally change this 
`require('exports?VanillaTilt!../../../../node_modules/vanilla-tilt/dist/vanilla-tilt.min.js');`
to this `require('vanilla-tilt');` :D

Thanks a lot!

**Comment by BrunnerLivio** [comment]
I don't know how finicky @micku7zu is, but your commit messages are not written in the general [commit-style guide](https://github.com/erlang/otp/wiki/Writing-good-commit-messages). Maybe change the names to 'Update build output' and 'Fix build script and add prebuplish script'? But it's just a detail, keep up the good work!

**Comment by micku7zu** [comment]
Thanks guys for your work, really appreciate it.

Now, we are really close to something we all want, but we are not there yet.

@bastilimbach can you please read this thread from here: https://github.com/micku7zu/vanilla-tilt.js/pull/1#issuecomment-275900814

Can we achieve something like that?

**Comment by bastilimbach** [comment]
@BrunnerLivio micku7zu didn't provide any contribution or style guides so I thought this would be unnecessary. But you're right, I should definitely write better commits!

@micku7zu I'll take a look at that 👍

**Comment by bastilimbach** [comment]
I think this would accomplish that: 
```javascript
  /**
   * Module Wrapper
   * --------------
   */
  if (typeof define === 'function' && typeof define.amd === 'object' && define.amd) {
    define(function () {
      return ScrollReveal
    })
  } else if (typeof module !== 'undefined' && module.exports) {
    module.exports = ScrollReveal
  } else {
    window.ScrollReveal = ScrollReveal
  }
```
-- Copied from [ScrollReveal](https://github.com/jlmakes/scrollreveal)

This is a wrapper around the module export. It checks, if `module` is defined, if it is not, then the script is just included using `<script>` tags and it will "return" `window.ScrollReveal = ScrollReveal`. Otherwise, if the module is defined, then the script knows, it is bundled by Webpack etc.

**Comment by bastilimbach** [comment]
@micku7zu Ok the module wrapper worked better than expected. Tested in Browser and with `require(vanilla-tilt)`. Both worked fine.

@BrunnerLivio could you test if everything is still working after these commits?

File: src/vanilla-tilt.js [hunk 1]
```diff
@@ -202,3 +202,17 @@ if (typeof document !== "undefined") {
...
+/**
+ * Module Wrapper
+ * --------------
+ */
+if (typeof define === 'function' && typeof define.amd === 'object' && define.amd) {
+  define(function () {
+    return VanillaTilt
+  })
+} else if (typeof module !== 'undefined' && module.exports) {
+  module.exports = VanillaTilt
```
Comment (bastilimbach) [inline src/vanilla-tilt.js]: @micku7zu I couldn't write this in ES5 because babel doesn't know how to compile this.

**Comment by micku7zu** [comment]
Thank you for your help. I managed to publish it to NPM after 2 hours of getting "npm 500 error" when I tried to publish. The problem was the "dist" key from package.json. NO FEEDBACK FROM NPM WHY IT FAILS. Just 500 error.

Just look here: https://www.google.ro/search?%7Bgoogle:acceptedSuggestion%7Doq=npm+500+error&%7Bgoogle:instantFieldTrialGroupParameter%7Dsourceid=chrome&ie=UTF-8&q=npm+500+error&gws_rd=ssl#q=npm+publish+500+error

So many different errors for CODE E500. Problems about the login, about the already taken package name, no one ever said about that "dist" key from package.json.

I just debugged until I found it and finally published it to npm.

Thanks for your help

**Comment by bastilimbach** [comment]
@micku7zu Why did you closed the PR? The prepublish script and the module wrapper aren't merged yet.

**Comment by micku7zu** [comment]
I got a warning from npm that prepublish is deprecated in npm v5.

And for the wrapper, I will have a look on weekend 😄 

Thanks again @bastilimbach, really good job.

**Comment by micku7zu** [comment]
I need to check what is the best option, to have them separated like it's right now, or together using a wrapper :)

**Comment by bastilimbach** [comment]
Alright 👍  I'm always happy to help and I'm looking forward to implementing vanilla-tilt into my personal website with just a single `require`. 😀

**Pull request closed without merging**